src/SSVOpenHexagon/Core/HGUpdate.cpp

```diff
@@ -27,9 +27,18 @@ void HexagonGame::update(ssvu::FT mFT)
 
     // TODO: refactor to avoid repetition, and truncate floating point number
     // TODO: also show best record (here) and last run + best record (in menu)
-    steamManager.set_rich_presence_in_game(
-        levelData->name, status.getTimeSeconds());
-    steamManager.run_callbacks();
+
+    const int DELAY_TO_UPDATE = 5; // X seconds
+    static float timeTillUpdate = 0.0;
+    timeTillUpdate -= mFT / 60.0;
+
+    if (timeTillUpdate <= 0)
+    {
+        // only required for Steam
+        steamManager.set_rich_presence_in_game(levelData->name, status.getTimeSeconds());
+        steamManager.run_callbacks();
+        timeTillUpdate = DELAY_TO_UPDATE;
+    }
 
     discordManager.set_rich_presence_in_game(
```

Comment:
Doesn't Discord need the same treatment?

Reply:
Had a quick check by watching someone else play, and it does appear so. I will apply the same thing to discord and update the card/PR's description.